cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -269,6 +269,15 @@ public void scanApplications(Cursor cursor,
         StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD);
   }
 
+  public List<ApplicationMeta> getAllApplications() throws IOException {
```

Comment:
What is this for? This is dangerous for performance since it is getting all applications. Generally we shouldn't have method like this for possible misuse.

Reply:
This function returns application details for all applications across all namespaces (existing function `getAllApplications(String namespaceId)` gets application details for applications in a given namespace).
It is used to get a list of all deployed plugins, from which we compute the plugin count metric.

Yes, this is an expensive operation that's done when:
- a pipeline is deployed
- a pipeline is deleted
- appfabric startup

Is there a more efficient way to get plugin counts?